src/analysis/functions.rs

```diff
@@ -206,14 +210,32 @@ fn analyze_function(
     configured_functions: &[&config::functions::Function],
     imports: &mut Imports,
 ) -> Info {
-    let async = func.parameters.iter().any(|parameter| parameter.scope == ParameterScope::Async);
+    let async = func.name.ends_with("_async") ||
+                func.parameters.iter()
+                               .any(|parameter| parameter.scope == ParameterScope::Async &&
+                                                parameter.c_type == "GAsyncReadyCallback");
+    let expecting_data = !async &&
+                         func.parameters.iter()
+                                        .any(|par| par.c_type.ends_with("Func") ||
+                                                   par.c_type.ends_with("Callback"));
+
     let mut commented = false;
     let mut bounds: Bounds = Default::default();
     let mut to_glib_extras = HashMap::<usize, String>::new();
     let mut used_types: Vec<String> = Vec::with_capacity(4);
     let mut trampoline = None;
+    let mut callbacks = Vec::new();
+    let mut destroys = Vec::new();
     let mut async_future = None;
 
+    if !async && !expecting_data &&
+       func.parameters.iter().any(|par| par.c_type == "GDestroyNotify") {
+        warn!("Function \"{}\" with destroy callback without callbacks", func.name);
```

Comment:
Maybe add a note here what this actually means, and one or two examples :) I assume this is for things that take a `GDestroyNotify` to handle the lifetime of a non-closure parameter?

Reply:
It's litteraly what it says: there is a destroy callback and there are no other callbacks.

Reply:
Ok, I meant more in words that are not what the code is doing but what this actually means API-wise.